### ISSUE TITLE
interp: Regression on $PATH handling on Windows

### ISSUE THREAD

**Issue by andreynering** [mvdan/sh#768]
Originally reported at:

- https://github.com/go-task/task/issues/619
- https://github.com/arduino/arduino-cli/pull/1576

Didn't had the time to try it myself yet (sorry), but wanted to report this sooner than later.

Apparently the latest release is messing with how executables see $PATH on Windows. For example, when running `go build foo.go` Gosh has no problems finding `go`, but the `go` command has problems finding other executables like `gcc`.

**Comment by mvdan** [comment]
If anyone can bisect to a single commit, or produce a small reproducer, that would certainly help :)

**Comment by glumia** [comment]
Hi @mvdan! (and thanks @andreynering for reporting the issue here)

Here is a minimal reproducer:
```
.
└── main.go
```

main.go:
```go
package main

import (
	"fmt"
	"os"
)

func main(){
	fmt.Println(os.Getenv("PATH"))
}
```

Commands (on Powershell):
```powershell
> go install mvdan.cc/sh/v3/cmd/gosh@v3.4.0
> gosh -c 'go run main.go'
C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Go\bin;C:\TDM-GCC-64\bin;C:\Program Files\Git\cmd;C:\Users\gius\AppData\Local\Microsoft\WindowsApps;C:\Users\gius\go\bin
> go install mvdan.cc/sh/v3/cmd/gosh@v3.4.1
> gosh -c 'go run main.go'
C:\Windows\system32:C:\Windows:C:\Windows\System32\Wbem:C:\Windows\System32\WindowsPowerShell\v1.0\:C:\Windows\System32\OpenSSH\:C:\Program Files\Go\bin:C:\TDM-GCC-64\bin:C:\Program Files\Git\cmd:C:\Users\gius\AppData\Local\Microsoft\WindowsApps:C:\Users\gius\go\bin
```

It seems like the issue are those `:` used as separators instead of `;`.

**Comment by glumia** [comment]
Uh, interesting! 

> Apologies for the trouble.

You don't have to apologize, you don't owe us anything.
Thanks for your work on this project and for looking into the issue in such a short time :)

**Comment by mvdan** [comment]
I have cherry-picked the fix into the v3.4 branch: https://github.com/mvdan/sh/commit/5ae9d64e14020598d3f40f94d7a4e2a27d3083a0

I'll do a v3.4.2 release sometime in the next week, to wait and see if any other bugs are spotted. In the meantime, you can use the HEAD of that branch, since master may include v3.5 feature work.

**Comment by andreynering** [comment]
Thanks @mvdan!
